core/thread/inc/TPool.h

```diff
@@ -0,0 +1,218 @@
+// @(#)root/thread:$Id$
+// Author: Xavier Valls March 2016
+
+/*************************************************************************
+ * Copyright (C) 1995-2006, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TPOOL
+#define ROOT_TPOOL
+
+#include "TCollection.h"
+#include "ROOT/TSeq.h"
+
+template< class F, class... T>
+ using noReferenceCond = typename std::enable_if<"Function can't return a reference" && !(std::is_reference<typename std::result_of<F(T...)>::type>::value)>::type;
```

Comment:
Does this using statement needs to be in the global state?
It does not follow the name conventions.
It seems to be a duplicate from TProcPool.h